### PR TITLE
Unit tests for onboarding modal

### PR DESCRIPTION
**Pull request by bionara**
# Description
Unit test(s) for the onboarding modal. Added a nw function to the assessment coordinator to making testing easier, and also easier reading code where we need reported_by_another . Hopefully the tests provide some useful use-cases for mocking such responses.

[comment]: <> (Description of the feature or fix)
[comment]: <> (Include link to Figma designs, if relevant)
[comment]: <> (Include link to Notion ticket, if relevant)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx [hunk 1]
```diff
@@ -8,27 +10,58 @@ import { Provider as ReduxProvider } from 'react-redux';
...
+    const elementBase = <MockedNavigator Component={CovidTestListScreen} />;
+    const store = mockStore({ content: { startupInfo: { show_covid_test_onboarding: true } } });
+    const elementWithRedux = <ReduxProvider store={store}>{elementBase}</ReduxProvider>;
+    const element = <ThemeProvider theme={theme}>{elementWithRedux}</ThemeProvider>;
+    const instance = renderer.create(element).root;
+    const modal = instance.findByType(CovidTestListOnboardingModal);
+    expect(modal).toBeTruthy();
+  });
+
+  it('does not show onboarding modal whn flag is false', () => {
```
Comment (juliatan) [inline src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx:46]: Small typo: 'when'

**Review by juliatan — APPROVED** [review]
This is so great @bionara !

File: src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx (hunk 1 above)
Comment (bionara) [reply src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx]: Fixed thanks!

File: src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx [hunk 2]
```diff
@@ -8,27 +10,58 @@ import { Provider as ReduxProvider } from 'react-redux';
 import renderer from 'react-test-renderer';
 import createMockStore from 'redux-mock-store';
 import { ThemeProvider } from 'styled-components/native';
-
-import { initialState } from '../../../../__mocks__/mockedInitialState';
 import MockedNavigator from '../../../../__mocks__/MockedNavigator';
```
Comment (selvach) [inline src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx:13]: Is absolute imports preferred?

File: src/core/assessment/AssessmentCoordinator.ts [hunk 3]
```diff
@@ -242,6 +242,11 @@ export class AssessmentCoordinator extends Coordinator {
   resetVaccine() {
     this.assessmentData.vaccineData = undefined;
   }
+
+  isReportedByOther() {
+    const isReportedByOther = this.assessmentData?.patientData?.patientInfo?.reported_by_another;
```
Comment (selvach) [inline src/core/assessment/AssessmentCoordinator.ts:247]: 💭   Unnecessary const. Just return. But thanks for wrapping this!

File: src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx (hunk 2 above)
Comment (aikewoody) [reply src/features/covid-tests/__tests__/CovidTestListScreen.test.tsx:13]: @selvach Yes!

**Review by selvach — APPROVED** [review]

**Pull request merged**